src/Specification/Truthy.php

```diff
@@ -17,6 +17,17 @@ public function isSatisfiedBy($value) : bool
     {
         // Note that a loose comparison ($value == true) is not used as it is too loose.
 
+        $value = $this->extract($value);
+
         return true === $value || 1 === $value || '1' === $value;
     }
+
+
+    /**
+     * Override this method to extract the value which will be checked for falsyness.
+     */
+    protected function extract($value)
```

Comment:
s: You could provide this through an abstract class.

Reply:
Most of the time this will want to have a specific return type hint so there's probably not mutch value in an abstract class. A trait might be useful though, as well as for common return types like `string`.